script/core/AssetManager.js

```diff
@@ -0,0 +1,118 @@
+import SpriteAtlas from 'core/SpriteAtlas';
+import FileManager from 'core/FileManager';
+
+const SPRITE = 'SPRITE';
+
+const assets = new Map();
+let spriteIndex;
+
+export default class AssetManager {
+	static async loadSpriteIndex(url) {
+		spriteIndex = await FileManager.loadFile(url, 'json')
+	}
+
+	static getAsset(name, type) {
+		if (!assets.has(name)) {
+			if (type === SPRITE) {
```

Comment:
Als er meer types assets komen straks ga je hier meer if's plaatsen? :)

Zo lang `AssetManager` niks anders doet dan sprites zou ik het gewoon `SpriteLoader` of whatever noemen en de implementatie specifiek houden. Dat je dit object nu al generieker opzet zou je premature optimization kunnen noemen :)

Reply:
Ja, supergoed, dat is voor nu veel beter.